torchvision/models/_utils.py

```diff
@@ -64,3 +67,127 @@ def forward(self, x):
+        This override:
+        1) Stores away the qualified name of the caller for restoration later
+        2) Installs the qualified name of the caller in `current_module_qualified_name`
+           for retrieval by `create_proxy`
+        3) Delegates into the normal Tracer.call_module method
+        4) Restores the caller's qualified name into current_module_qualified_name
+        """
+        old_qualname = self.current_module_qualified_name
+        try:
+            module_qualified_name = self.path_of_module(m)
+            self.current_module_qualified_name = module_qualified_name
+            if not self.is_leaf_module(m, module_qualified_name):
+                out = forward(*args, **kwargs)
+                self.node_to_originating_module[out.node] = module_qualified_name
+                return out
+            return self.create_proxy('call_module', module_qualified_name, args, kwargs)
+        finally:
+            self.current_module_qualified_name = old_qualname
+
+    def create_proxy(self, kind: str, target: torch.fx.node.Target, args: Tuple[Any, ...],
+                     kwargs: Dict[str, Any], name: Optional[str] = None, type_expr: Optional[Any] = None):
+        """
+        Override of `Tracer.create_proxy`. This override intercepts the recording
+        of every operation and stores away the current traced module's qualified
+        name in `node_to_originating_module`
+        """
+        proxy = super().create_proxy(kind, target, args, kwargs, name, type_expr)
+        self.node_to_originating_module[proxy.node] = self.current_module_qualified_name
+        return proxy
+
+
+def get_intermediate_layers(model: nn.Module, return_layers: Dict[str, str]) -> nn.Module:
+    """
+    Creates a new FX-based module that returns intermediate layers from a given model.
+    This is achieved by re-writing the computation graph of the model via FX to return
+    the requested layers.
+
+    All unused layers are removed, together with their corresponding parameters.
+
+    Args:
+        model (nn.Module): model on which we will extract the features
+        return_layers (Dict[name, new_name]): a dict containing the names
+            of the modules for which the activations will be returned as
+            the key of the dict, and the value of the dict is the name
+            of the returned activation (which the user can specify).
+
+    Examples::
+
+        >>> m = torchvision.models.resnet18(pretrained=True)
+        >>> # extract layer1 and layer3, giving as names `feat1` and feat2`
+        >>> new_m = torchvision.models._utils.get_intermediate_layers(m,
+        >>>     {'layer1': 'feat1', 'layer3': 'feat2'})
+        >>> out = new_m(torch.rand(1, 3, 224, 224))
+        >>> print([(k, v.shape) for k, v in out.items()])
+        >>>     [('feat1', torch.Size([1, 64, 56, 56])),
+        >>>      ('feat2', torch.Size([1, 256, 14, 14]))]
+    """
+    # TODO come up with a better name for this
```

Comment:
I think `return_layers` is fine. I understand it remaps but it still stores the mapping of the returned layers.